ts/src/pro/bitmart.ts

```diff
@@ -524,8 +525,7 @@ export default class bitmart extends bitmartRest {
                 ],
             };
             const message = this.extend (request, params);
-            future = this.watch (url, messageHash, message);
-            client.subscriptions[messageHash] = future;
```

Comment:
@pcriadoperez are you 100% sure that we should remove this line? Suppose the user calls authenticate 2-3 times in parallel.

Reply:
In this case we are checking the subscriptionHash to make sure it's not called several times.

It started from this discussion from these two PRs, that first Huobi was failing for php
- https://github.com/ccxt/ccxt/pull/18819/files
- https://github.com/ccxt/ccxt/pull/18812

And we decided on the patten of: 
- `authenticate` calls client.future
- We check the subscriptions if there is the messageHash of authenticated not to call it several times.
- In `handleAuthenticate` we manualy resolve the future instead of using client.resolve

Now I'm double checking the rest of the exchanges to follow the same pattern.